### ISSUE TITLE
Rendering issue in IE8b2 #50

### ISSUE THREAD

**Issue by ghost** [cdotyone/mochaui#2]
Reported by tjmcgrew, Oct 17, 2008

What steps will reproduce the problem?
1. Open the demo site with Internet Explorer 8 beta 2.

What is the expected output? What do you see instead?
The window decorations for the popup windows do not render. They do seem to
work correctly however. Also, the buttons on the right side of the taskbar
do not render correctly.

What version of MochaUI are you using?
0.9.5

On what operating system and with what browser?
Windows XP SP3, Internet Explorer 8 beta 2

Please provide any additional information below.
The fact that IE8 is still in beta would make this low priority, but
something to be aware of. I have also tried this using IE8's compatibility
mode, with the same results. I have attached a screenshot.

```
mochaui-ie8b2.jpg
```

153 KB   View   Download
Comment 2 by tjmcgrew, Oct 17, 2008

Apologies,

After some further research I see that you're aware of the problem and that the issue
is with IE8 itself.

Comment 3 by gregory.houston, Oct 17, 2008

Yeah, we are sort of in a holding pattern until the next IE8 release. Hopefully there
will be at least one more beta and they will have all the issues worked out in it,
though I wouldn't count on it.

MochaUI with Excanvas works in IE8 beta 2:
- Browser Mode: IE7, Document Mode: Quirks Mode

Caveats: Being in quirks mode causes other CSS problems. Not working in IE8 Standards
mode makes it so you cannot use the new CSS features implemented in IE8.

MochaUI with Moocanvas works in:
- Browser Mode: IE7, Document Mode: Quirks Mode
- Browser Mode: IE7, Document Mode: IE7 Standards Mode
- Browser Mode: IE8, Document Mode: Quirks Mode
- Browser Mode: IE8, Document Mode: IE7 Standards Mode
- Browser Mode: IE8 Compatibility View, Document Mode: Quirks Mode
- Browser Mode: IE8 Compatibility View, Document Mode: IE7 Standards Mode
  http://ibolmo.com/projects/moocanvas/

Caveats: There are some issues still with Moocanvas. The CoolClock widget example in
MochaUI does not display properly with it. Ibolmo, the author of Moocanvas is aware
of the clock widget problem. He may fix it, but no guarantees. Also, not working in
IE8 Standards mode makes it so you cannot use the new CSS features implemented in IE8.

For Excanvas or Moocanvas to work in IE8 Standards Mode will require the
Excanvas code be modified to use import instead of the of the
star selector (v: *).
https://connect.microsoft.com/IE/feedback/ViewFeedback.aspx?FeedbackID=333905

Also, the following bug will need to be fixed for either Excanvas or
Moocanvas to work in IE8 Standards Mode.
https://connect.microsoft.com/IE/feedback/ViewFeedback.aspx?FeedbackID=364096

So to sum up, we are waiting on a couple things, bug fixes in IE8, and then a new
release of either Excanvas or Moocanvas that uses import rather than the star selector.

In the worst case scenario, if Microsoft fails to fix the bugs and there is not a new
release of Excanvas that works in as many modes as Moocanvas, then I will make a new
release of MochaUI with Moocanvas in IE7 standards mode. The downside to this is that
it will break the clock widget and will also not allow developers to use new CSS
features added to IE8.

One other fix required in MochaUI for IE8 Standards Mode is to add another line in
the CSS wherever opacity is set:

```
-ms-filter: "progid:DXImageTransform.Microsoft.Alpha(Opacity=0)"; /* IE8 */     
filter: progid:DXImageTransform.Microsoft.Alpha(Opacity=0); /* IE6 and 7*/  
opacity: .0;
-moz-opacity: 0;
```

Comment 4 by gregory.houston, Dec 23, 2008

There is a new version of Excanvas included in r386. I had to tweak a little bit of
code in MochaUI to get it working but it seems to be okay. It is not compressed yet.

The clock is not currently working in IE8 beta 2. I will probably wait to fool with
it more until the IE8 release candidate is out.

So other than the clock the canvas elements are rendering properly in IE8 Standards
Mode. 

The window resize event does not fire in IE8 beta 2. This is a known issue and
supposedly is fixed in the current IE8 nightly build.

There are some strange errors here and there. Hopefully they will be resolved in the
release candidate. If not, I'll explore them further then.

Comment 5 by gregory.houston, Feb 08, 2009

As of r426 in the SVN, all of the MochaUI demos should now be
compatible with IE8 RC1 in both the "IE=7" and "IE=8" X-UA-Compatible
modes. You could also set the X-UA-Compatible meta tag to "IE=edge" or
remove it entirely.

I would suggest using "IE=8" unless you have your own code that breaks
in IE8 Standards Mode. If your code does break, then try "IE=7". If
your code breaks in both modes then you too will have the joy of
modifying your code to work in IE8 RC1.

SVN:
http://code.google.com/p/mocha-ui/source/checkout

Let me know if  you find any further bugs. Depending on what craziness
IE does between now and their final release they may very well create
more regression bugs that we will have to deal with after the IE8
final release.

As long as it isn't too far off in the future, sometime after the
final release of IE8 I will probably release MochaUI 1.0. So if you
want to get your sites IE8 compatible before that time you will
probably need to get the code from the SVN.

Also of interest, The IE8 Readiness Toolkit:
http://www.microsoft.com/windows/internet-explorer/beta/readiness/developers-new.aspx

I will probably wait to close this ticket until after IE8's final release.

Status: Accepted
Comment 6 by ged...@gmail.com, May 28, 2009

I am using the trunk (r526) here and in IE8 on vista it all works.
good work Greg !!

Comment 7 by webmaster1989, Jan 28, 2010

OW THX!!!! I loveeeeeeeeeee Mocha UI
